### ISSUE TITLE
Write better guidelines for metadata

### ISSUE THREAD

**Issue by Marigold** [owid/etl#158]
From discussion here https://github.com/owid/etl/pull/144#pullrequestreview-937631199

> Thanks @Marigold ! The data, metadata and pipeline look good. Just two comments (that should not affect this PR, but rather future pipelines):

> We could consider establishing some guidelines on what metadata should be for meadow, and what for garden or grapher. In this case, I think it is a bit arbitrary. Most metadata could be from the beginning in meadow (except the note about Tanzania, which is relevant only for >= garden steps).

> It's not clear what should happen next time we update this dataset. Should we duplicate the garden/ggdc/2020-10-01/ folder and its content, and replace the date "2020-10-01" everywhere? It seems a bit redundant, although it's not too worrying, given that this will happen at most once per year for this dataset.

We should update README with proper guidelines as soon as we agree on what should they be.

**Comment by pabloarosado** [comment]
Thanks for starting this, @Marigold .
Regarding metadata, I suppose the logic should be to aim at adding all possible metadata as far back as possible (ideally walden). And any piece of metadata that becomes relevant only after a certain step should be added at that specific step (e.g. "This region has been ignored due to...").
A possible downside of this is that most metadata would end up being stored inside the "ingests" folder of walden, which doesn't sound like the place where it was meant to be.

**Comment by Marigold** [comment]
@pabloarosado yes, that's exactly the dilemma I was facing for Maddison. Another issue is that Walden's metadata has way different structure than ETL metadata, so it's often easier to add it in Walden. Maybe we should even think about better alignment of Walden and ETL (similar metadata, access through shared catalog) to make our lives easier.

**Comment by larsyencken** [comment]
My original thinking doesn't quite lock down where the metadata gets added. It went something like:

- `walden` is for snapshots, and we should have some minimum level of metadata there (e.g. including the license for data), but you should be able to add a snapshot in 15 minutes without thinking too hard about it, even if you later come back to improve it.
- `meadow` is for _syntactic harmonisation_, meaning getting something into the right format. The idea is that instead of sitting on the dataset that you're working on for a long time without sharing it, you could spend a few hours and get it into a form where others can already reuse it. Out of this you get more, smaller pull-requests. It's also important to have this form for debugging, so that we can compare changes we make later to the "original data".
- `garden` is for _semantic harmonisation_, meaning the field names and field values are harmonised whenever possible. So all datasets in garden should play well together, you should be able to join across datasets easily and without accidents. At the same time, this is the place where we want to add metadata standards.

The reason we didn't add metadata requirements for `garden` is that we were trying to move towards a situation where the data team could stop working in long-term branches and instead work more iteratively. So the idea was that as long as it's harmonised it can merged into garden; that everything in garden gets published; but we try not to leave data long term in garden without adding metadata.

I tend to agree Pablo that earlier is better, where possible. If it fits in Walden's metadata model, it should go there. But I wouldn't add extra metadata to Walden, I'd add it in a later step.

I'd be pretty happy with a convention that said: 

1. Fill out `walden` metadata as best you can, if it can be later improved, do it there. 
2. Do everything else in the `garden` step.
3. Do anything grapher-specific last-minute in the grapher step.

Thoughts?

**Comment by pabloarosado** [comment]
Hi @larsyencken I think your logic makes sense, although I am thinking of one way to solve the problem of not knowing where to put the metadata. The slightly different logic would be:
* The code in `walden` (for example in `ingests`) aims at gathering snapshots of raw data from different sources.
  * We don't add any metadata to `walden`. There, we simply have raw snapshots of the data.
  * Every snapshot in `walden` requires at least one dataset in `etl` `meadow`, where the metadata is added.
* The code in `etl` aims at processing data. Given that writing metadata in a convenient format is a way of processing raw data, then it should be a step of `etl`. And the logic of where to put metadata would be as follows:
  * Metadata added in `meadow` is related to intrinsic features of dataset (e.g. license).
  * Metadata added in `garden` is related to the processing that OWID does (e.g. country standardization).

This logic has the benefit that all metadata stays in `etl`, and we don't have ambiguity on where to add it. However, the small downside is that it forces us to always have a `meadow` step. This is additional work, but it also makes the existence of `meadow` more meaningful (which so far to me it wasn't very clear yet).

What do you think?

**Comment by Marigold** [comment]
> Fill out walden metadata as best you can, if it can be later improved, do it there.

This was a bit of a pain for Maddison - we had description in walden and any updates to it meant extra commit in walden. It'd be easier to work just within ETL. Anyway, **this is not a big deal**, I don't think we need any special process for this. As long as the metadata ends up in garden, it's fine. I personally prefer to have all metadata in garden step (even extra metadata for grapher like `display`) because it's all in one place.

**Comment by larsyencken** [comment]
I'm looking at the Maddison `description` in Walden. It seems like a massive "how to cite this dataset" block, but it does not actually describe the dataset. This makes me wonder if the original description was just not written well enough, or whether we have a metadata alignment issue. A consumer outside of grapher who stumbled across this dataset using our tools would want to know more about it before deciding whether to use it.

On the more general note of Walden though, I think the repo isn't useful without the metadata. We're actually republishing the data just by mirroring it, so already the license and source info become important immediately. If the hassle of having a separate repo for it is too much, let's just merge walden into the ETL codebase so that we don't have to keep crossing the submodule boundary for our work.

**Comment by Marigold** [comment]
We just copied the description from the original dataset and moved it as upstream as possible (walden). This wasn't a great idea, we should have written a separate description for walden and then use this one in garden / grapher. Still learning ;). I think there was some confusion about which metadata should go to walden, now it's clear. So to sum it up, there's nothing wrong with walden, we just used it inappropriately.

> If the hassle of having a separate repo for it is too much, let's just merge walden into the ETL codebase

That's tempting, but it would end up as a mess since we store the index in git.
